de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -477,15 +477,15 @@ protected void awaitActivityQueueingActivation(IProgressMonitor monitor)
       throws SarosCancellationException {
 
     monitor.beginTask(
-        "Waiting for " + getPeer().getName() + " to continue the project negotiation...",
+        "Waiting for " + getRemoteUser() + " to continue the project negotiation...",
```

Comment:
You should not expect that the toString() method of objects deliver something that is meant to be read by a normal user. So the correct replacement is getRemoteUser().getJID().getName()

Reply:
With retrieve I was for now just looking at `jid.getName()`, but yes XmppConnection in User is not a good idea. So probably set the display name in the CTOR and unify the whole sometimes the GUI is displaying the jid and sometimes a nick name.